cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/trigger/TriggerJsonCodec.java

```diff
@@ -22,16 +22,27 @@
 import com.google.gson.JsonObject;
 import com.google.gson.JsonParseException;
 import com.google.gson.JsonPrimitive;
+import com.google.gson.JsonSerializationContext;
+import com.google.gson.JsonSerializer;
 
 import java.lang.reflect.Type;
 
 /**
  * Serialization and deserializtion of Triggers as Json.
  *
- * All triggers inherit the {@link Trigger#className} field from the base class. Here we use that
- * during deserialization to determine the actual subclass, and then deserialize based on that.
+ * All triggers must be serialized by passing in the {@link Trigger} class as the type of the object.
```

Comment:
"All triggers must be serialized by passing in the {@link Trigger} class as the type of the object"
What if a Trigger is a field of an object X, and we want to serialize that object X? We can't pass in the Trigger class as the type of the object in that case, right?